src/main/scala/nl.knaw.dans.easy.umd/UpdateMetadataDataset.scala

```diff
@@ -61,24 +61,33 @@ object UpdateMetadataDataset extends DebugEnhancedLogging {
       transformer = Transformer(record.streamID, record.xmlTag, record.oldValue, record.newValue)
       newXML = transformer.transform(oldXML)
       _ <- reportChanges(record, oldXML, newXML)
-      _ <- fedora.updateDatastream(record.fedoraID, record.streamID, newXML.toString())
+      _ <- fedora.updateDatastream(record.fedoraID, record.streamID, stringify(newXML))
     } yield ()
   }.recoverWith {
-    case e => Failure(new Exception(s"failed to process: $record, reason: ${e.getMessage}", e))
+    case e => Failure(new Exception(s"failed to process: $record, reason: ${ e.getMessage }", e))
   }
 
-  private def reportChanges(record: InputRecord, oldXML: Elem, newXML: Seq[Node]): Try[Unit] = {
-    if (oldXML == newXML)
-      Failure(new Exception(s"could not find ${record.streamID} <${record.xmlTag}>${record.oldValue}</${record.xmlTag}>"))
-    else Try {
-      val oldLines = new PrettyPrinter(160, 2).format(oldXML).linesIterator.toList
-      val newLines = new PrettyPrinter(160, 2).format(newXML.head).linesIterator.toList
-
-      logger.info(s"old ${record.streamID}: ${compare(oldLines, newLines)}")
-      logger.info(s"new ${record.streamID}: ${compare(newLines, oldLines)}")
+  private def reportChanges(record: InputRecord, oldXML: Elem, newXML: Seq[Node]): Try[Unit] = Try {
```

Comment:
Regarding this method: is it true that if either one of the `logger.error` conditions is met, the rest of the procedure should still continue? What I mean is: above you see `reportChanges` being called in a for-comprehension. Should `fedora.updateDatastream` be called if `reportChanges` ends up in the `logger.error` cases? If not, please return a `Failure`, as was the case in the previous implementation.

Reply:
The intention is that these commands (adding or replacing) are idempotent.  It was so that if  the intended replacement/addition had already taken place the execution stopped.  I think it shouldn't.  The other commands in the given input file may well be valid.  Now we just get errors in the log file (errors in order to draw more attention to what happened) and we go on.